src/dialect/postgresql.rs

```diff
@@ -67,6 +85,102 @@ impl Dialect for PostgreSqlDialect {
         )
     }
 
+    fn get_next_precedence(&self, parser: &Parser) -> Option<Result<u8, ParserError>> {
```

Comment:
@alamb @samuelcolvin : wouldn't it be better to have a `Result<Option<u8>>` instead of an `Option<Result<u8>>`

Reply:
that would:
* differ from the signature of `Iterator::next()`
* be a breaking change to the API

Reply:
I also think the `transpose()` method can be used to convert `Option<Result<..>>` to `Result<Option<..>>` 

https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose

I don't have any particular preference for one over the other